src/supervisor-api.ts

```diff
@@ -13,17 +13,24 @@ import { checkTruthy } from './lib/validation';
 import log from './lib/supervisor-console';
 
 function getKeyFromReq(req: express.Request): string | null {
-	const queryKey = req.query.apikey;
-	if (queryKey != null) {
-		return queryKey;
+	// Check query for key
+	if (req.query.apikey) {
+		return req.query.apikey;
 	}
-	const maybeHeaderKey = req.get('Authorization');
-	if (!maybeHeaderKey) {
+	// Get Authorization header to search for key
+	const authHeader = req.get('Authorization');
+	// Check header for key
+	if (!authHeader) {
 		return null;
 	}
-
-	const match = maybeHeaderKey.match(/^ApiKey (\w+)$/);
-	return match != null ? match[1] : null;
+	// Check authHeader with various schemes
+	return ['ApiKey', 'Bearer'].reduce((apiKey, scheme) => {
```

Comment:
_.find would be clearer here

Reply:
I think it'd make more sense to match what was here before:
```suggestion
	const match = authHeader.match(/^(?:ApiKey|Bearer) (\w+)$/);
	return match?.[1];
```
although it might also make sense to make it case-insensitive to make life easier:
```suggestion
	const match = authHeader.match(/^(?:ApiKey|Bearer) (\w+)$/i);
	return match?.[1];
```